python/revng/scripting/projects/daemon_project.py

```diff
@@ -0,0 +1,192 @@
+#
+# This file is distributed under the MIT License. See LICENSE.md for details.
+#
+
+import json
+from typing import Any, List
+
+import yaml
+from gql import Client, gql
+from gql.transport.aiohttp import AIOHTTPTransport
+
+from revng.model import Binary, dump_diff_to_string, load_model_from_string, make_diff
+
+from .project import Project
+
+
+class DaemonProject(Project):
+    def __init__(self, url: str):
+        self.transport = AIOHTTPTransport(url=f"{url}/graphql/")
+        self.client = Client(
+            transport=self.transport, fetch_schema_from_transport=True, execute_timeout=None
+        )
+        super().__init__()
+
+    def import_binary(self, input_binary_path: str) -> None:
+        query = gql(
+            """
+            mutation upload($file: Upload!) {
+                uploadFile(file: $file, container: "input")
+            }
+        """
+        )
+
+        with open(input_binary_path, "rb") as binary_file:
+            result = self.client.execute(
+                query, variable_values={"file": binary_file}, upload_files=True
+            )
+
+        if not result["uploadFile"]:
+            raise RuntimeError("File upload failed")
+
+    def get_artifact(self, artifact_name: str) -> bytes | str:
+        return self._get_artifact_internal(artifact_name)
+
+    def _get_artifact_internal(self, artifact_name: str, targets_class: List[Any] = []) -> bytes | str:
+        if not targets_class:
+            targets_class = self.model.Functions
+        targets = [tc.key() for tc in targets_class]
+
+        result = self._get_artifact(artifact_name, targets)
+        return self._get_result_mime(artifact_name, result)
+
+    def _get_artifact(self, artifact_name: str, targets_class: List[str] = []) -> bytes | str:
+        query = gql(
+            """
+            query($step: String!, $paths: String!, $index: BigInt!) {
+                produceArtifacts(step: $step, paths: $paths, index: $index) {
+                    ... on Produced {
+                        result
+                    }
+                }
+            }
+        """
+        )
+        variables = {
+            "step": artifact_name,
+            "paths": ",".join(targets_class),
+            "index": self._get_content_commit_index(),
+        }
+
+        response = self.client.execute(query, variable_values=variables)
+        return response["produceArtifacts"]["result"].encode()
+
+    def analyze(self, analysis_name: str, target_class: list[Any] = []) -> None:
```

Comment:
`-> None` isn't needed, drop it.

Reply:
By this I assume you mean to remove all the `-> None` where the method doesn't return anything? If that is the case, wouldn't it be better to always set the return `type` just to be consistent? Or is there another reason to remove it?